tools/socket_passing.py

```diff
@@ -13,45 +13,65 @@
 import httplib
 import json
 import os.path
+import re
 import sys
 import time
+import yaml
 
 # Seconds to wait for the admin address output file to appear. The script exits
 # with failure if the file is not found.
 ADMIN_FILE_TIMEOUT_SECS = 20
 
-def GenerateNewConfig(original_json, admin_address, updated_json):
+# Because the hot restart files are yaml but yaml support is not included in
+# python by default, we parse this fairly manually.
+def GenerateNewConfig(original_yaml, admin_address, updated_json):
   # Get original listener addresses
-  with open(original_json, 'r') as original_json_file:
-    # Import original config file in order to get a deterministic output. This
-    # allows us to diff the original config file and the updated config file
-    # output from this script to check for any changes.
-    parsed_json = json.load(original_json_file, object_pairs_hook=OrderedDict)
-  original_listeners = parsed_json['listeners']
-
-  sys.stdout.write('Admin address is ' + admin_address + '\n')
-  try:
-    admin_conn = httplib.HTTPConnection(admin_address)
-    admin_conn.request('GET', '/listeners')
-    admin_response = admin_conn.getresponse()
-    if not admin_response.status == 200:
-      return False
-    discovered_listeners = json.loads(admin_response.read())
-  except Exception as e:
-    sys.stderr.write('Cannot connect to admin: %s\n' % e)
-    return False
-  else:
-    if len(discovered_listeners) != len(original_listeners):
+  with open(original_yaml, 'r') as original_file:
+    sys.stdout.write('Admin address is ' + admin_address + '\n')
+    try:
+      admin_conn = httplib.HTTPConnection(admin_address)
+      admin_conn.request('GET', '/listeners')
+      admin_response = admin_conn.getresponse()
+      if not admin_response.status == 200:
+        return False
+      discovered_listeners = json.loads(admin_response.read())
+    except Exception as e:
+      sys.stderr.write('Cannot connect to admin: %s\n' % e)
       return False
-    for discovered, original in zip(discovered_listeners, original_listeners):
-      if discovered.startswith('/'):
-        original['address'] = 'unix://' + discovered
-      else:
-        original['address'] = 'tcp://' + discovered
-    with open(updated_json, 'w') as outfile:
-      json.dump(OrderedDict(parsed_json), outfile, indent=2, separators=(',',':'))
-  finally:
-    admin_conn.close()
+    else:
+      raw_yaml = original_file.readlines()
```

Comment:
Sorry why do we need to do this? Can't we use a venv with some better parsing stuff?

Reply:
Is there a compromise where we don't need any venv but just do a couple of very dumb regexes?

Reply:
We can do a venev, but this script is called under blaze test //test/integration:all.  I think spinning up a virtual environment and installing yaml is way too heavyweight to do on every single test run. 
 Beyond being slow and expensive it'd also make tests fail if you're running them locally (with no network connectivity).  I really don't think it's the way to go, but htuch@ feels as strongly that we shouldn't add yaml to the docker build image because of second order effects with version conflicts and didn't like the original json verison either.  Options as they stand are

1) venv (alyssa dislikes)
2) add yaml to our docker image (htuch dislikes)
3) use json config for this test (htuch dislikes)
4) parse yaml manually (matt appears to dislike)

Surely we can find some compromise we all dislike evenly?  :-D

@htuch this was the version where I didn't venv and I do dumb regex.    I just forgot to remove the yaml import (now fixed)

Reply:
The reason not to add the yaml package to the Docker environment is that this forces every developer who builds natively to put this in their Python system library. We really don't want any non-hermetic dependencies. You could try make it a Bazel native dependency, but I suspect that will be more effort than it's worth, unless @jmillikin-stripe could jump in and explain how to do it.

I'm surprised how complicated the "dumb regex" method is; shouldn't it be just a few lines?

Reply:
not when we're replacing 7 addresses and I'm writing the python.

Reply:
OK, I think this code is just duct tape test infra anyway, I don't feel that strongly about making it super clean. I think potentially more verbose code than the minimal needed is less worse than using JSON as example human readable configs, so ship it.